src/main/java/org/rulez/demokracia/pdengine/VoteRegistry.java

```diff
@@ -15,7 +15,14 @@ public VoteRegistry(WebServiceContext wsContext) {
 	@Override
 	public String obtainBallot(String id, String adminKey) {
 		Vote vote = getVote(id);
-		vote.checkAdminKey(adminKey);
+		
+		if(adminKey.equals("anon")) {
+			if(!vote.countedAssurances.equals(vote.neededAssurances))
```

Comment:
should be logically vote.neededAssurances.containsAll(user.assurances)
use hasAssurance (and mock wsContext in the test)

Reply:
Sorry I do not get it clearly.
So I get the user with getWsUserName() but it is just a user name.
How can I get the user object with the assurances param?

Reply:
You do not get an user object here. It is too far in the dependency tree. You can check whether a user has an assurance with hasAssurance().
Actually we do not yet have a User object, I have just put that into the design, not yet committed.
Once we will have one, this can be solved like this:
SessionFactory manager could have a getAssurances() method, which casts the wsContext to our wsContext implementation, and that implementation could have its own getAssurances()method, which return the assurances of the underlying User object. But none of it exists yet. You can choose to implement it and mock your way through it, or just use hasAssurance() for each of the  neededAssurances.